### PR TITLE
Run clang-tidy modernize checks

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR just blindly runs `run-clang-tidy-12.py -checks="-*,modernize-*" -header-filter=.* -fix`. Manually revered changes in tinymt32.h and STLTuple.h. Manually revert some refactorings resulting in compile errors.

### CODE REVIEW

File: include/alpaka/core/ConcurrentExecPool.hpp [hunk 1]
```diff
@@ -174,7 +172,7 @@ namespace alpaka
 
             private:
                 //! The execution function.
-                virtual auto run() -> void final
```
Comment (j-stephan) [inline include/alpaka/core/ConcurrentExecPool.hpp:177]: Why is the `virtual` removed here?

**Review by j-stephan — DISMISSED** [review]

File: include/alpaka/core/ConcurrentExecPool.hpp (hunk 1 above)
Comment (fwyzard) [reply include/alpaka/core/ConcurrentExecPool.hpp:177]: both `override` and `final` imply `virtual`, so `clang-tidy` removes it

Comment (j-stephan) [reply include/alpaka/core/ConcurrentExecPool.hpp:177]: Ah, I see. Thanks!

**Comment by j-stephan** [comment]
Some clang versions don't like these changes:

https://gitlab.com/hzdr/crp/alpaka/-/pipelines/460211464

**Comment by bernhardmgruber** [comment]
> Some clang versions don't like these changes:
> 
> https://gitlab.com/hzdr/crp/alpaka/-/pipelines/460211464

I already fixed these today, but the GitLab CI is not restarting.

**Comment by j-stephan** [comment]
Hm, it could be that you need to force-push again. I think because of today's GitLab issues the pipeline wasn't notified about anything you committed after yesterday.

**Review by j-stephan — APPROVED** [review]

**Pull request merged**